cmd/export.go

```diff
@@ -16,24 +16,101 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
+	"strings"
+
 	export "github.com/pesos/grofer/src/export/general"
 	"github.com/spf13/cobra"
 )
 
+const (
+	DefaultExportRefreshRate = 1000
+	DefaultExportIterations  = 10
+	DefaultExportFileName    = "grofer_profile"
+	DefaultExportType        = "json"
+)
+
+func validateFileName(fileName string, exportType string) error {
```

Comment:
@MadhavJivrajani Do you think enforcing file names is a good idea? I feel would be better if the user could name the exported file whatever he/she pleases.

Reply:
The user can name it as they please. The constants are to handle un-specified parameters.

Reply:
Say in the case when a user provides `myprofile` and `json` as filenames and export types, the if condition on line 70 would fail and this will then cause the `validateFileName()` function to fail too.

I felt we could drop the appending of extension entirely as this might cause issues when users attempt to script with grofer and the filenames change. But if necessary to append a file extension, we could probably check for the export type as a suffix and act on that accordingly.

What do you think?

Reply:
The if condition on line 70 should fail right, we want a filename with an extension. The appending is done only in case of default file name (the case where the user does not specify a file name). The appending is not done always.

> But if necessary to append a file extension, we could probably check for the export type as a suffix and act on that accordingly.

That's what's being done. The suffix (last element of the `split` slice (split based on `.`)) is validated, including the case of the user not specifying the extension at all.

> as this might cause issues when users attempt to script with grofer and the filenames change.

Could you give an example of this? Not too sure about what you mean.

Reply:
> Could you give an example of this? Not too sure about what you mean.

Consider this shell script.

```sh
# You can ignore the below command execution, just need the var.
FILENAME=$(sed 's/old_text/new_text/g' some_file | grep "some pattern") 
grofer export --type json --filename $FILENAME
cat $FILENAME | grep "something"
```

In the current case, the user would probably have to use `cat ${FILENAME}.json` in the last line

> That's what's being done. The suffix (last element of the split slice (split based on .)) is validated, including the case of the user not specifying the extension at all.

How are you handling this? If no extension is given, the filename will be sent to `validateFileName()` and there the check will fail right?

Reply:
Wait so, you want it to allow filenames that don't have extensions? 
I wrote the checks to enforce that every file provided (except the one that is unspecified) has an extension and a valid one (corresponding to the passed export type arg). IMO it would be better to let the user handle doing `cat ${FILENAME}.json` rather than creating a file with no extension.

Reply:
Hmm actually I see your point now, I guess it would be better to leave providing an extension for the file to the user rather than enforcing it.

Reply:
> Wait so, you want it to allow filenames that don't have extensions?

Yes, I figured it would be better to print it out exactly as requested.

Reply:
If adamant about appending file name, maybe display a message saying "Succesfuly exported to file.json" maybe?

Reply:
@Gituser143 I've made a few changes. Let me know what you think. I've retained appending an explicit type in the default case since if a file name is not specified then a valid extension being present in the filename would ensure lack of any ambiguity.

Reply:
@Gituser143 I've made a few changes. Let me know what you think.

Reply:
> If adamant about appending file name, maybe display a message saying "Succesfuly exported to file.json" maybe?

I've removed that aspect completely In the current commit